tests/migrations/test_operations.py

```diff
@@ -182,3 +187,57 @@ def test_modify_ttl() -> None:
         ).format_sql()
         == "ALTER TABLE test_table REMOVE TTL;"
     )
+
+
+def test_specify_order() -> None:
+    """
+    Test that specifying the migration order works when changing forwards_local_first
+    """
+    create_local_op = Mock(CreateTable)
+    create_dist_op = Mock(CreateTable)
+    drop_local_op = Mock(DropTable)
+    drop_dist_op = Mock(DropTable)
+    logger = Logger("test")
+    context = Context("001", logger, lambda x: None)
+
+    class TestMigration(migration.ClickhouseNodeMigration):
+        blocking = False
+
+        def forwards_local(self) -> Sequence[SqlOperation]:
+            return [create_local_op]
+
+        def backwards_local(self) -> Sequence[SqlOperation]:
+            return [drop_local_op]
+
+        def forwards_dist(self) -> Sequence[SqlOperation]:
+            return [create_dist_op]
+
+        def backwards_dist(self) -> Sequence[SqlOperation]:
+            return [drop_dist_op]
+
+    ops = [create_local_op, create_dist_op, drop_local_op, drop_dist_op]
+    order = []
+    for op in ops:
+
+        def effect(op: Mock) -> Callable[[bool], None]:
+            def add_op(local: bool) -> None:
+                order.append(op)
+
+            return add_op
+
+        op.execute.side_effect = effect(op)
+
+    test_migration = TestMigration()
+    test_migration.forwards(context)
+    assert order == [create_local_op, create_dist_op]
+    order.clear()
+    test_migration.backwards(context, False)
```

Comment:
why does the `backwards` method need the extra arg?

Reply:
not sure why, but on our migrations `dry_run` isn't optional for backwards. surprised me too.

Reply:
hmm my guess is that's an oversight, but if it isn't thats not super clear haha. okay then we can leave the test as is and find out if we should be passing a default value of `False` for backwards as well